torch/nested/_internal/ops.py

```diff
@@ -472,7 +472,15 @@ def to_copy_default(func, *args, **kwargs):
 
     new_values = func(inp._values, **new_kwargs)
     new_offsets = inp._offsets.to(device=new_values.device)
-    _tensor_symint_registry[new_offsets] = _tensor_symint_registry[inp._offsets]
+
+    from torch._subclasses.fake_tensor import FakeTensor
+    from torch._subclasses.functional_tensor import FunctionalTensor
+
+    if isinstance(new_offsets, (FakeTensor, FunctionalTensor)):
+        # Temporary hack until we have the union find
+        new_offsets.nested_int_memo = inp._offsets.nested_int_memo
```

Comment:
I was just looking around trying to understand why we need to define a `nested_into_memo` and `get_nested_int` in `FunctionalTensor` directly - is this the reason?

instead of having FunctionalTensor know about nested_int directly, I think I would rather have the nested code that needs to get/set a `nested_int` just know about FunctionalTensor and do any necessary unwrapping (it also needs to know about `FunctionalTensor` in the isinstance check above). Does that seem reasonable?

Reply:
It actually looks like the aot autograd calling view_as to produce grad_outputs logic does not run with FunctionalTensors. Updated here and the constructor to unwrap as needed.